airflow-core/src/airflow/models/connection.py

```diff
@@ -498,8 +497,8 @@ def get_connection_from_secrets(cls, conn_id: str) -> Connection:
                     SecretCache.save_connection_uri(conn_id, conn.get_uri())
                     return conn
             except Exception:
-                log.debug(
-                    "Unable to retrieve connection from secrets backend (%s). "
+                log.exception(
+                    "Error while retrieving connection from secrets backend (%s). "
```

Comment:
```suggestion
                    "Unable to retrieve connection from secrets backend (%s). "
```

for past consistency